tests/Microsoft.Identity.Test.Unit/AppConfigTests/ManagedIdentityApplicationBuilderTests.cs

```diff
@@ -80,27 +81,29 @@ public void TestConstructor_WithCreateUserAssignedId()
         }
 
         [DataTestMethod]
-        [DataRow(TestConstants.ClientId)]
-        [DataRow("resourceId", false)]
-        [DataRow("resourceId/subscription", false)]
-        public void TestConstructor_WithUserAssignedManagedIdentity_ResourceId(string userAssignedId, bool isClientId = true)
+        [DataRow(TestConstants.ClientId, UserAssignedIdentityId.ClientId)]
+        [DataRow("resourceId", UserAssignedIdentityId.ResourceId)]
+        [DataRow("resourceId/subscription", UserAssignedIdentityId.ResourceId)]
+        [DataRow(TestConstants.ObjectId, UserAssignedIdentityId.ObjectId)]
+        public void TestConstructor_WithUserAssignedManagedIdentity_ResourceId(string userAssignedId, UserAssignedIdentityId userAssignedIdentityId)
         {
-            var mi = ManagedIdentityApplicationBuilder.Create(isClientId ? 
-                    ManagedIdentityId.WithUserAssignedClientId(userAssignedId) : 
-                    ManagedIdentityId.WithUserAssignedResourceId(userAssignedId))
-                .BuildConcrete();
+            ManagedIdentityApplication mi = CreateMIABuilder(userAssignedId, userAssignedIdentityId).BuildConcrete();
 
             Assert.AreEqual(userAssignedId, mi.ServiceBundle.Config.ClientId);
             Assert.IsNotNull(mi.ServiceBundle.Config.ManagedIdentityId);
 
-            if (isClientId)
+            if (userAssignedIdentityId == UserAssignedIdentityId.ClientId)
```

Comment:
This `if` can just be one:
`Assert.AreEqual(userAssignedIdentityId, mi.ServiceBundle.Config.ManagedIdentityId.IdType);`

Reply:
updated